src/internal/event-bridge.ts

```diff
@@ -0,0 +1,139 @@
+import { JSONObject } from 'k6'
+import http, { RefinedResponse, ResponseType } from 'k6/http'
+
+import { AWSClient } from './client'
+import { AWSConfig } from './config'
+import { AWSError } from './error'
+import { InvalidSignatureError, SignatureV4 } from './signature'
+import { AMZ_TARGET_HEADER } from './constants'
+import { HTTPHeaders, HTTPMethod } from './http'
+
+type PutEventEntry = {
+    Detail: string
+    DetailType: string
+    EventBusName: string
+    Resources: [string]
+    Source: string
+}
+
+type PutEventsInput = {
+    EndpointId?: string
+    Entries: PutEventEntry[]
```

Comment:
Based on the fact that the [`PutEventsRequestEntry` type](https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_PutEventsRequestEntry.html) defines all its fields as optional, let's use the following definition instead:

```suggestion
    Entries: Partial<PutEventEntry>[]
```

So that it's possible to effectively pass in a `PutEventEntry` with partially undefined properties.

Reply:
That link does however state that `Source`, `Detail` and `DetailType` are required fields to send events to Event Bridge, but the others aren't. So if you're in agreement, I'll set these as required, but the others as optional.

Reply:
This is my bad. That will teach me to not read the excerpt with the blue background 🤓 My understanding was based on the `Required: no` statements under each field. 

That makes sense, then, and let's go forward with setting only the non-necessary fields as optional 🙇🏻